test/modules/TwinTester/TwinEdgeOperationsInitializer.cs

```diff
@@ -17,41 +18,56 @@ class TwinEdgeOperationsInitializer : ITwinTestInitializer
 
-        TwinEdgeOperationsInitializer(RegistryManager registryManager, ModuleClient moduleClient, ITwinTestResultHandler reporter, TwinState initializedState)
+        TwinEdgeOperationsInitializer(RegistryManager registryManager, ModuleClient moduleClient, ITwinTestResultHandler reporter, int reportedPropertyUpdateCounter)
         {
-            this.reportedPropertyUpdater = new ReportedPropertyUpdater(registryManager, moduleClient, reporter, initializedState);
-            this.desiredPropertiesReceiver = new DesiredPropertyReceiver(registryManager, moduleClient, reporter);
+            this.registryManager = registryManager;
+            this.reportedPropertyUpdater = new ReportedPropertyUpdater(moduleClient, reporter, reportedPropertyUpdateCounter);
+            this.desiredPropertiesReceiver = new DesiredPropertyReceiver(moduleClient, reporter);
         }
 
         public static async Task<TwinEdgeOperationsInitializer> CreateAsync(RegistryManager registryManager, ModuleClient moduleClient, ITwinTestResultHandler reporter)
         {
             try
             {
-                TwinState initializedState;
-                Twin twin = await registryManager.GetTwinAsync(Settings.Current.DeviceId, Settings.Current.ModuleId);
+                Twin twin = await registryManager.GetTwinAsync(Settings.Current.DeviceId, Settings.Current.TargetModuleId);
 
-                // reset properties
-                Twin desiredPropertyResetTwin = await registryManager.ReplaceTwinAsync(Settings.Current.DeviceId, Settings.Current.ModuleId, new Twin(), twin.ETag);
-                initializedState = new TwinState { TwinETag = desiredPropertyResetTwin.ETag };
+                // reset reported properties
+                TwinCollection resetProperties = TwinTesterUtil.GetResetedReportedPropertiesTwin(twin);
+                await moduleClient.UpdateReportedPropertiesAsync(resetProperties);
 
-                Logger.LogInformation($"Start state of module twin: {JsonConvert.SerializeObject(twin, Formatting.Indented)}");
-                return new TwinEdgeOperationsInitializer(registryManager, moduleClient, reporter, initializedState);
+                return new TwinEdgeOperationsInitializer(registryManager, moduleClient, reporter, 0);
             }
             catch (Exception e)
             {
                 throw new Exception($"Shutting down module. Initialization failure: {e}");
             }
         }
 
-        public async Task Start()
+        public async Task StartAsync(CancellationToken cancellationToken)
         {
+            await Task.Delay(Settings.Current.TestStartDelay, cancellationToken);
+            await this.DisplayInitialTwin();
             this.periodicUpdate = new PeriodicTask(this.UpdateAsync, Settings.Current.TwinUpdateFrequency, Settings.Current.TwinUpdateFrequency, Logger, "TwinReportedPropertiesUpdate");
             await this.desiredPropertiesReceiver.UpdateAsync();
         }
 
-        public void Dispose()
```

Comment:
why remove this method? should this class implement IDisposable since it owns periodic task which implements IDisposable. right?

Reply:
I changed the method to stop because it should not dispose the initializer, but just stop the periodictask. After the test completes it should continue to receive desired properties updates.